src/dune_tui/dune_tui.ml

```diff
@@ -144,35 +33,304 @@ module Tui = struct
     fun () -> !(Lazy.force setup)
   ;;
 
-  let start () = ()
+  (** style for diving visual elements like borders or rules *)
+  let divider_attr = A.(fg red)
+
+  (** style for helpful ui elements like scrollbar structures or help text *)
+  let helper_attr = A.(fg yellow)
+
+  (** style for user feedback like message count, or scrollbar position *)
+  let user_feedback_attr = A.(fg cyan)
+
+  (** [horizontal_rule ~attr ~w] draws a horizontal line with [attr] of length [w]. *)
+  let horiztonal_rule ~attr ~w = I.uchar attr Unicode.horizontal_bar w 1
+
+  (** Here we keep some persistent state about the program that we udpate each time we
+      render. This allows other components to "react" to changes using [Lwd]. *)
+
+  let term_size = Lwd.var (0, 0)
+  let messages = Lwd.var []
+  let status_line = Lwd.var None
+
+  module Message_viewer = struct
+    (** Specialized widget for viewing messages in a list. *)
+
+    (** [is_message_hidden] tracks if a given message is hidden. [true] means it is hidden
+        and [false] means it is expanded. *)
+    let is_message_hidden = Lwd.var (fun _ -> false)
 
-  let image ~status_line ~messages =
+    (** We share some work turning messages into images. *)
```

Comment:
I don't understand this comment

Reply:
We are just sharing this Lwd between a few places. If we inline it, we would compute the value twice. I will get rid of the comment since it is not so helpful.